pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -86,73 +252,45 @@ func (os *OpenStack) DetachDisk(instanceID string, partialDiskId string) error {
 		glog.Errorf("Unable to initialize nova client for region: %s", os.region)
 		return err
 	}
-	if len(disk.Attachments) > 0 && disk.Attachments[0]["server_id"] != nil && instanceID == disk.Attachments[0]["server_id"] {
+	if volume.AttachedServerId != instanceID {
+		errMsg := fmt.Sprintf("Disk: %s has no attachments or is not attached to compute: %s", volume.Name, instanceID)
+		glog.Errorf(errMsg)
```

Comment:
@anguslees, thank you for the review, that is a direct port of logic of previous code. If/when time comes to add v3 support I will keep this in mind.